src/app/core/profile/researcher-profile.service.ts

```diff
@@ -0,0 +1,266 @@
+import { Item } from '../shared/item.model';
+import { NoContent } from '../shared/NoContent.model';
+import {
+  getFinishedRemoteData,
+  getFirstCompletedRemoteData,
+  getFirstSucceededRemoteDataPayload
+} from '../shared/operators';
+import { ResearcherProfile } from './model/researcher-profile.model';
+import { RESEARCHER_PROFILE } from './model/researcher-profile.resource-type';
+import { HttpOptions } from '../dspace-rest/dspace-rest.service';
+import { PostRequest } from '../data/request.models';
+import { hasValue } from '../../shared/empty.util';
+import {CoreState} from '../core-state.model';
+
+/**
+ * A private DataService implementation to delegate specific methods to.
+ */
+class ResearcherProfileServiceImpl extends DataService<ResearcherProfile> {
+  protected linkPath = 'profiles';
+
+  constructor(
+    protected requestService: RequestService,
+    protected rdbService: RemoteDataBuildService,
+    protected store: Store<CoreState>,
+    protected objectCache: ObjectCacheService,
+    protected halService: HALEndpointService,
+    protected notificationsService: NotificationsService,
+    protected http: HttpClient,
+    protected comparator: DefaultChangeAnalyzer<ResearcherProfile>) {
+    super();
+  }
+
+}
+
+/**
+ * A service that provides methods to make REST requests with researcher profile endpoint.
+ */
+@Injectable()
+@dataService(RESEARCHER_PROFILE)
+export class ResearcherProfileService {
+
+  dataService: ResearcherProfileServiceImpl;
+
+  responseMsToLive: number = 10 * 1000;
+
+  constructor(
+    protected requestService: RequestService,
+    protected rdbService: RemoteDataBuildService,
+    protected store: Store<CoreState>,
+    protected objectCache: ObjectCacheService,
+    protected halService: HALEndpointService,
+    protected notificationsService: NotificationsService,
+    protected http: HttpClient,
+    protected router: Router,
+    protected comparator: DefaultChangeAnalyzer<ResearcherProfile>,
+    protected itemService: ItemDataService,
+    protected configurationService: ConfigurationDataService ) {
+
+    this.dataService = new ResearcherProfileServiceImpl(requestService, rdbService, store, objectCache, halService,
+      notificationsService, http, comparator);
+
+  }
+
+  /**
+   * Find the researcher profile with the given uuid.
+   *
+   * @param uuid the profile uuid
+   */
+  findById(uuid: string): Observable<ResearcherProfile> {
+    return this.dataService.findById(uuid, false)
+      .pipe ( getFinishedRemoteData(),
+        map((remoteData) => remoteData.payload));
+  }
+
+  /**
+   * Create a new researcher profile for the current user.
+   */
+  create(): Observable<RemoteData<ResearcherProfile>> {
+    return this.dataService.create( new ResearcherProfile());
+  }
+
+  /**
+   * Delete a researcher profile.
+   *
+   * @param researcherProfile the profile to delete
+   */
+  delete(researcherProfile: ResearcherProfile): Observable<boolean> {
```

Comment:
Same comment as findByID above, though if I'm not mistaken this change stems from #1617 and I already commented on it there